src/uct/cuda/cuda_copy/cuda_copy_iface.h

```diff
@@ -48,23 +49,30 @@ typedef struct uct_cuda_copy_queue_desc {
     ucs_queue_elem_t            queue;
 } uct_cuda_copy_queue_desc_t;
 
+typedef struct uct_cuda_copy_per_ctx_rsc {
+    CUcontext                   cuda_ctx;
+    unsigned long long          ctx_id;
+    /* pool of cuda events to check completion of memcpy operations */
+    ucs_mpool_t                 cuda_event_desc;
+    /* stream used to issue short operations */
+    CUstream                    short_stream;
+    /* array of queue descriptors for each src/dst memory type combination */
+    uct_cuda_copy_queue_desc_t  queue_desc[UCS_MEMORY_TYPE_LAST][UCS_MEMORY_TYPE_LAST];
+} uct_cuda_copy_per_ctx_rsc_t;
+
+
+KHASH_MAP_INIT_INT64(cuda_copy_ctx_rscs, struct uct_cuda_copy_per_ctx_rsc*);
```

Comment:
you can store `uct_cuda_copy_per_ctx_rsc` (not the pointer), then you would need to do alloc/free during put.

Reply:
@brminich Will incorporate this change.